Add font-lock for el-patch-def* forms

Fuco1 opened the pull request:
Fixes #35

Fuco1 commented:
Yea that makes sense.  I'm leaving for a vacation now for 10 days so I'll cook something afterwards.

raxod502 commented:
Sounds good. Enjoy your trip!

Fuco1 commented:
Hey @raxod502 I've done some changes, is this in line with what you invisioned?

I'll add a documentation entry after the review.

raxod502 commented on el-patch.el:
```diff
@@ -651,48 +675,55 @@ DEFINITION is a list starting with `defun' or similar."
 (el-patch-deftype defconst
   :classify el-patch-classify-variable
   :locate el-patch-locate-variable
+  :font-lock #'el-patch-fontify-as-variable
```
For consistency, I'd like these to be unquoted.

Fuco1 replied:
Fair enough.  I personally prefer as little magic as possible especially with macros, so I pass quoted or unquoted based on how it is used later.  But consistency almost always trumps purity :)

raxod502 replied:
Yes, I think you are right about preferring quoted, and if I were designing this macro again then I would do it your way. It's not a good enough reason to break backwards compatibility, though :P

raxod502 commented:
This thread is being closed automatically by [Tidier](https://github.com/raxod502/tidier) because it is labeled with "waiting on response" and has not seen any activity for 90 days. But don't worry—if you have any information that might advance the discussion, leave a comment and the thread may be reopened :)

Fuco1 commented:
Hey, I completely forgot about this not being merged, as I use it in my config since forever.

I've updated the `:font-lock` to be passed unquoted and rebased to the latest HEAD.  When you have a moment check it out! :)

Fuco1 commented:
(seems like github doesn't re-check my repo if this is closed, maybe reopening will make it refresh... as of now it still shows the old patch)

raxod502 commented:
Unfortunately, it seems like it's not possible to re-open the pull request, because I have since deleted the `develop` branch of `el-patch`. Would you mind opening a new pull request with the same change?
